changes.go

```diff
@@ -0,0 +1,37 @@
+package sgreplicate
+
+type ChangedRev struct {
+	Revision string `json:"rev"`
+}
+
+type Change struct {
+	Sequence    interface{}  `json:"seq"`
+	Id          string       `json:"id"`
+	ChangedRevs []ChangedRev `json:"changes"`
+	Removed     []string     `json:"removed"`
+	Deleted     bool         `json:"deleted"`
+}
+
+type Changes struct {
+	Results      []Change    `json:"results"`
+	LastSequence interface{} `json:"last_seq"`
+}
+
+// If any changes entries have a "removed" property, ignore them.
+// See https://github.com/couchbase/sync_gateway/issues/2212 for details.
+func filterRemovedChanges(changes Changes) Changes {
+	newChanges := Changes{}
+	newChanges.LastSequence = changes.LastSequence
+
+	newChanges.Results = []Change{}
+	for _, change := range changes.Results {
+		if change.Removed != nil && len(change.Removed) > 0 {
+			// This doc was removed from channel, ignore it
+			continue
+		}
```

Comment:
It doesn't look like this handles the case where the doc is visible via to the changes request via more than one channel, and it's only removed from one of them.  In that scenario removed will be non-nil, but we'd want to replicate the doc.

Here's a sample changes response for a user with access to channels ABC, DEF, where rev-1 of document `doc` was in both channels, and rev-2 is only in DEF.
```
{
    "last_seq": "22", 
    "results": [
        {
            "changes": [
                {
                    "rev": "3-51daf08bf1a9a87deb1e7d9f57d31bb9"
                }
            ], 
            "id": "doc", 
            "removed": [
                "ABC"
            ], 
            "seq": 22
        }
    ]
}
```